### ISSUE TITLE
Is dumb-jump expected to honour is dumb-jump-selector via xref-find-definitions evals? 

### ISSUE THREAD

**Issue by pierre-rouleau** [jacktasia/dumb-jump#370]
Now that dumb-jump promotes the use of xref backend by adding dumb-jump-xref-activate to the ref-backend-functions this enables the use of the `M-.` key binding to perform a identifier jump.   However when doing so, dumb-jump does not honour the dumb-jump-selector user-option: the prompt is done by the xref mechanism.

* Is this the intention?
* If not, is there something else that needs to be done in the init.el configuration code?
* If the question to the above 2 question is "no", is there a way to solve this without changing dumb-jump code?  

I know this is somewhat related to issue #365  but it's also something different (although the root cause might be the same).

**Comment by phikal** [comment]
It is intentional, as the point to using xref is to separate the logic and presentation. As the backend, it's only role is to find the results, while xref is responsible for user interaction.

Also see ["Obsolete commands and options"](https://github.com/jacktasia/dumb-jump#obsolete-commands-and-options), `dumb-jump-selector` is listed under the obsolete variables.

**Comment by pierre-rouleau** [comment]
OK, thanks.  Do you know how to use the presentation provided by the obsoleted dumb-jump-selector while using the xref backend?

**Comment by phikal** [comment]
That would be like placing the cart before the horse, if you're using xref, xref decides how to display it's results.

What you could look into, is if someone created custom functions for `xref-show-xrefs-function` and `xref-show-definitions-function`. AFAIK those two options should modify how the xref UI behaves.

**Comment by pierre-rouleau** [comment]
Yep, that's what I'm doing.  Out of curiosity, why were the dumb-jump-selector infrastructure and the functions that use it like dumb-jump-go obsoleted?  I understand that it's a good thing to use the xref system, but I would have thought that one would want to fully integrate the functionality of dumb-jump with xref in doing so.

**Comment by phikal** [comment]
Most of the discussion on that question can be found in  #349 and related issues.

The short version is basically to avoid having to actively maintain two separate user-facing interfaces, that could potentially restrict how the internals could evolve.

**Comment by pierre-rouleau** [comment]
I have written a function that can be used as the ``xref-show-xrefs-function`` but i'm still missing something to get it working: the ``xref-show-xrefs-function``is given a list of xref objects that contain the path and line, I can deduct the context, but I don't have the original identifier that is used by the dumb-jump code as the ``:target`` property.  I have not yet found anything where I get extract the identifier that was passed to the original ``xref-find-definition`` evaluation.   Short of looking into the call stack I don't see anything at the moment.  Any idea?

I'll also start looking for changing the xref result presentation handling independently of dumb-jump as its done in some packages.
